archaius2-core/src/main/java/com/netflix/archaius/instrumentation/AccessMonitorUtil.java

```diff
@@ -108,8 +109,8 @@ private void flushUsageData() {
 
     /** Merge the results of given accessMonitorUtil into this one. */
     public void merge(AccessMonitorUtil accessMonitorUtil) {
-        for (Map.Entry<String, PropertyUsageData> entry : accessMonitorUtil.propertyUsageMap.entrySet()) {
-            propertyUsageMap.putIfAbsent(entry.getKey(), entry.getValue());
+        for (Map.Entry<String, PropertyUsageData> entry : accessMonitorUtil.propertyUsageMapRef.get().entrySet()) {
+            propertyUsageMapRef.get().putIfAbsent(entry.getKey(), entry.getValue());
```

Comment:
Grab a reference to the destination map before entering the loop and merge the entries into that. Otherwise, if a call to `getAndClear` comes in while the loop is running then half of the other `accessMonitorUtil`'s entries will be merged into the old map and half into the new, which will be confusing at best :-P

Reply:
Oops yea good point. I think in practice this should never happen, but good to safeguard against it